deps/build.jl

```diff
@@ -109,8 +109,20 @@ println("Generating gap.sh ...")
 
 gap_sh_path = joinpath(gap_bin_root, "bin", "gap.sh")
 write(gap_sh_path,"""
-#!/bin/sh
-
-exec "$(gap_bin_root)/gap" -l "$(extra_gap_root);$(gap_src_root)" "\$@"
+#!/bin/bash
+# This is a a Julia script which also is a valid bash script; if executed
+# by bash, it'll execute itself via julia.
+#=
+export GAP_SHOW_BANNER=true
+exec julia --color=yes --startup-file=no "\${BASH_SOURCE[0]}" "\$@"
```

Comment:
Here's a problem with that: it hardcodes `julia` here :/. I guess I could make this dynamic by using `Sys.BINDIR` here; but then it'd be specific on the Julia version that was used to compile it. Which is reasonable, but it *could* break.

Then again, the old approach by launching the `gap` exe suffered from the exact same problem as it had to hardcode the rpath of libjulia; so I guess using Sys.BINDIR it is for now.

BTW, note that this `gap.sh` file actually is *also* a valid Julia script, so one could invoke it that way, too. 

However, we now are running into a bit of a chicken-and-egg kind of problem: I suggested to e.g. @mohamed-barakat to use `$(julia -e 'import GAP; print(GAP.gap_exe())')` to invoke the GAP-in-GAP.jl; well, we could change that to  `julia --color=yes --startup-file=no PATH/TO/gap.sh  "#$@"` except that we don't know `PATH/TO/gap.sh`... to figure that out ... we need to load GAP.jl. But if we do that, we are loading it "in the wrong way", so we can't just go back the REPL mode, and we can't just pass arguments to it... HRMM

Perhaps I can tweak things so that  we *can* enter the GAP prompt after loading GAP.jl "normally". For that I'd need at least...
- a way to reset the GAP error handler (should be done in the GAP kernel, but I can also hack it into JuliaInterface for now
- reset `ERROR_OUTPUT` to its standard value
- ... ?

As an added benefit, we could then perhaps also add a nice `GAP.prompt()` proper around `GAP.Globals.SESSION()` which would do the same?

In the meantime, though, I noticed more issue with this PR:
- ctrl-C crashes; we ought to add `--handle-signals=no` to the arguments -- but then somehow ctrl-C does nothing anymore?
- something is broken with input / line editing
- ... needs much more testing ...

Reply:
I added `--handle-signals=no`, and also reactivated ctrl-c handling in GAP, and now handling `SIGINT` in GAP-via-Julia works on Linux. It is still broken on macOS, though. I thought that it was perhaps doing something macOS/Darwin/Mach-O specific, but I see nothing. I did notice that Julia has a dedicated thread for handling signals, including SIGINT, which uses `kqueue`. So perhaps that supersedes the signal handling in GAP? But then why only on macOS, not on Linux? Unclear to me. Perhaps @rbehrends has an idea?

Reply:
Another issue: starting this "GAP-in-Julia" sometimes segfaults. Unfortunately this happens very infrequently which makes it difficult to debug.

Reply:
There have been a lot of changes to `SIGINT` handling in Julia, which make it difficult to track down the exact behavior; basically, the [logic was changed](https://github.com/JuliaLang/julia/pull/16174) to deliver `SIGINT` only during GC safepoints (deferring them until then) and that affected overall signal handling logic. Also, with pthreads there are technically no guarantees which thread gets a signal, which may also result in system-specific differences. This is something I'll have to look into.